pkg/splunk/enterprise/util.go

```diff
@@ -1615,3 +1646,42 @@ func validateMonitoringConsoleRef(c splcommon.ControllerClient, revised *appsv1.
 	//if the sts doesn't exists no need for any change
 	return nil
 }
+
+// setInstallStateForClusterScopedApps sets the install state for cluster scoped apps
+func setInstallStateForClusterScopedApps(appDeployContext *enterpriseApi.AppDeploymentContext, status enterpriseApi.AppPhaseStatusType) {
+	scopedLog := log.WithName("setInstallStateForClusterScopedApps")
+
+	for appSrcName, appSrcDeployInfo := range appDeployContext.AppsSrcDeployStatus {
+		deployInfoList := appSrcDeployInfo.AppDeploymentInfoList
+		for i := range deployInfoList {
+			appSrc, err := getAppSrcSpec(appDeployContext.AppFrameworkConfig.AppSources, appSrcName)
+			if err != nil {
+				// Error, should never happen
+				scopedLog.Error(err, "Unable to find App src", "App src name", appSrcName, "App name", deployInfoList[i].AppName)
+				continue
+			}
+
+			if appSrc.Scope == enterpriseApi.ScopeCluster {
```

Comment:
Since the mapping is `appSrcName` to `deployInfoList`, we can do the scope check above line# 1655